### PR TITLE
Fixes issue #71 (adding new sections to the contributing guide)

### PR DESCRIPTION
**Pull request by HaimHorowitzAgnostiq**
- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation, VERSION, and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -455,6 +455,34 @@ Some important commands to know about:
...
 This module can be used to mock module variables, environment variables etc. Check out the official [monkeypatch documentation](https://docs.pytest.org/en/6.2.x/monkeypatch.html).
 
+## How to add new modules to the API
+
+## Opening and moving between branches
+
+To open a new branch:
+
+```bash
+git branch branch_name
```
Comment (FyzHsn) [inline CONTRIBUTING.md:465]: This should be:

When beginning implementation on an issue, first create a branch off of develop:

```bash
git branch branch_name develop
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -455,6 +455,34 @@ Some important commands to know about:
...
+```
+
+To move to a given branch:
+
+```bash
+git checkout branch_name
+```
+
```
Comment (FyzHsn) [inline CONTRIBUTING.md:473]: Add the next two lines:

The last two commands can be combined into one command:

```bash
git checkout -b branch_name develop
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -455,6 +455,34 @@ Some important commands to know about:
...
+## Linking Pool Request to issue number
```
Comment (FyzHsn) [inline CONTRIBUTING.md:474]: *Pull request

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -455,6 +455,34 @@ Some important commands to know about:
...
+
+In order to link a Pool Request to a given issue number, the first commit should be accompanied with a comment containing the issue number, for example:
```
Comment (FyzHsn) [inline CONTRIBUTING.md:476]: *Pull request

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -455,6 +455,34 @@ Some important commands to know about:
...
+
+```bash
+git commit -m 'fixes issue #57'
```
Comment (FyzHsn) [inline CONTRIBUTING.md:479]: Will mentioned to me that his needs to be (leave out the issue part): 

```bash
git commit -m 'fixes #57'
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -455,6 +455,34 @@ Some important commands to know about:
...
+```
+
+## How and when to do a Pull Request
+
+Pull requests should be done when work on an issue is done and ready to be reviewed. Once the new changes are pushed to remote, an option to do a Pull Request will appear on the top of the page at AgnostiqHQ/covalent. You will then be able to choose a reviewer on the upper right part of the screen.
```
Comment (FyzHsn) [inline CONTRIBUTING.md:484]: Not true. 

Pull requests can be done as soon as the implementation has started in a branch. Reviewers should be tagged when the changes are ready to be reviewed. 

I'd also say take the part about where the Pull requests will appear and also where one can select a reviewer. It's too basic for this contributing guide

**Review by FyzHsn — CHANGES_REQUESTED** [review]
Could you create a section called:

Getting started with feature development
===============================

at the top between the First steps and Style guide sections. 

And finally, move the info on creating branches, doing pull reviews etc to that section.

File: CONTRIBUTING.md (hunk 2 above)
Comment (FyzHsn) [reply CONTRIBUTING.md:473]: Add this in line 473.

**Comment by FyzHsn** [comment]
@HaimHorowitzAgnostiq I've added the latest changes. This is ready for your review.

@Emmanuel289 Adding you to this review as well. Let me know if there are any questions.

**Comment by Emmanuel289** [comment]
> @HaimHorowitzAgnostiq I've added the latest changes. This is ready for your review.
> 
> @Emmanuel289 Adding you to this review as well. Let me know if there are any questions.

@FyzHsn Thanks! I'll add in the list of new package installs on my Mac.

**Comment by Emmanuel289** [comment]
> > @HaimHorowitzAgnostiq I've added the latest changes. This is ready for your review.
> > @Emmanuel289 Adding you to this review as well. Let me know if there are any questions.
> 
> @FyzHsn Thanks! I'll add in the list of new package installs on my Mac.

@HaimHorowitzAgnostiq @FyzHsn  After running ```pip install -e .```,  the packages installed were 

```Successfully built e cova
Installing collected packages: e, cova
  Attempting uninstall: cova
    Found existing installation: cova 0.26.1
    Uninstalling cova-0.26.1:
      Successfully uninstalled cova-0.26.1
Successfully installed cova-0.27.2 e-1.4.5
```
After running ```pip install -r tests/requirements```, the following packages were installed:
```
Successfully built autograd future
Installing collected packages: webencodings, semantic-version, nodeenv, mistune, mccabe, ipython-genutils, iniconfig, distlib, appdirs, testpath, scipy, pyzmq, pyrsistent, pyflakes, pycodestyle, py, pluggy, pandocfilters, nest-asyncio, mock, jupyterlab-pygments, jupyter-core, isort, identify, future, filelock, entrypoints, defusedxml, coverage, cfgv, autoray, attrs, virtualenv, pytest, jupyter-client, jsonschema, flake8, bleach, autograd, pytest-mock, pytest-cov, pre-commit, pennylane, nbformat, nbclient, nbconvert
Successfully installed appdirs-1.4.4 attrs-21.4.0 autograd-1.3 autoray-0.2.5 bleach-4.1.0 cfgv-3.3.1 coverage-6.3.2 defusedxml-0.7.1 distlib-0.3.4 entrypoints-0.4 filelock-3.6.0 flake8-3.9.2 future-0.18.2 identify-2.4.11 iniconfig-1.1.1 ipython-genutils-0.2.0 isort-5.7.0 jsonschema-4.4.0 jupyter-client-7.1.2 jupyter-core-4.9.2 jupyterlab-pygments-0.1.2 mccabe-0.6.1 mistune-0.8.4 mock-4.0.3 nbclient-0.5.11 nbconvert-6.3.0 nbformat-5.1.3 nest-asyncio-1.5.4 nodeenv-1.6.0 pandocfilters-1.5.0 pennylane-0.16.0 pluggy-1.0.0 pre-commit-2.13.0 py-1.11.0 pycodestyle-2.7.0 pyflakes-2.3.1 pyrsistent-0.18.1 pytest-6.2.5 pytest-cov-2.12.0 pytest-mock-3.6.1 pyzmq-22.3.0 scipy-1.8.0 semantic-version-2.6.0 testpath-0.6.0 virtualenv-20.13.2 webencodings-0.5.1

```

**Comment by FyzHsn** [comment]
> > > @HaimHorowitzAgnostiq I've added the latest changes. This is ready for your review.
> > > @Emmanuel289 Adding you to this review as well. Let me know if there are any questions.
> > 
> > 
> > @FyzHsn Thanks! I'll add in the list of new package installs on my Mac.
> 
> @HaimHorowitzAgnostiq @FyzHsn After running `pip install -e .`, the packages installed were
> 
> ```
> Installing collected packages: e, cova
>   Attempting uninstall: cova
>     Found existing installation: cova 0.26.1
>     Uninstalling cova-0.26.1:
>       Successfully uninstalled cova-0.26.1
> Successfully installed cova-0.27.2 e-1.4.5
> ```
> 
> After running `pip install -r tests/requirements`, the following packages were installed:
> 
> ```
> Successfully built autograd future
> Installing collected packages: webencodings, semantic-version, nodeenv, mistune, mccabe, ipython-genutils, iniconfig, distlib, appdirs, testpath, scipy, pyzmq, pyrsistent, pyflakes, pycodestyle, py, pluggy, pandocfilters, nest-asyncio, mock, jupyterlab-pygments, jupyter-core, isort, identify, future, filelock, entrypoints, defusedxml, coverage, cfgv, autoray, attrs, virtualenv, pytest, jupyter-client, jsonschema, flake8, bleach, autograd, pytest-mock, pytest-cov, pre-commit, pennylane, nbformat, nbclient, nbconvert
> Successfully installed appdirs-1.4.4 attrs-21.4.0 autograd-1.3 autoray-0.2.5 bleach-4.1.0 cfgv-3.3.1 coverage-6.3.2 defusedxml-0.7.1 distlib-0.3.4 entrypoints-0.4 filelock-3.6.0 flake8-3.9.2 future-0.18.2 identify-2.4.11 iniconfig-1.1.1 ipython-genutils-0.2.0 isort-5.7.0 jsonschema-4.4.0 jupyter-client-7.1.2 jupyter-core-4.9.2 jupyterlab-pygments-0.1.2 mccabe-0.6.1 mistune-0.8.4 mock-4.0.3 nbclient-0.5.11 nbconvert-6.3.0 nbformat-5.1.3 nest-asyncio-1.5.4 nodeenv-1.6.0 pandocfilters-1.5.0 pennylane-0.16.0 pluggy-1.0.0 pre-commit-2.13.0 py-1.11.0 pycodestyle-2.7.0 pyflakes-2.3.1 pyrsistent-0.18.1 pytest-6.2.5 pytest-cov-2.12.0 pytest-mock-3.6.1 pyzmq-22.3.0 scipy-1.8.0 semantic-version-2.6.0 testpath-0.6.0 virtualenv-20.13.2 webencodings-0.5.1
> ```
> 
> After running `pip install -r requirements.txt` in `/covalent/docs` (Building RTD locally), the following packages were installed:
> 
> ```
> Successfully built autodocsumm sphinx-gallery lazy-object-proxy wrapt
> Installing collected packages: snowballstemmer, alabaster, wrapt, unidecode, sphinxcontrib-serializinghtml, sphinxcontrib-qthelp, sphinxcontrib-jsmath, sphinxcontrib-htmlhelp, sphinxcontrib-devhelp, sphinxcontrib-applehelp, soupsieve, pbr, mdurl, lazy-object-proxy, imagesize, docutils, babel, Sphinx, markdown-it-py, beautifulsoup4, astroid, sphinxcontrib-apidoc, sphinx-togglebutton, sphinx-prompt, sphinx-panels, sphinx-inline-tabs, sphinx-gallery, sphinx-copybutton, sphinx-click, sphinx-automodapi, sphinx-autodoc-typehints, sphinx-autoapi, mdit-py-plugins, furo, autodocsumm, myst-parser, nbsphinx
> Successfully installed Sphinx-4.0.3 alabaster-0.7.12 astroid-2.9.3 autodocsumm-0.2.6 babel-2.9.1 beautifulsoup4-4.10.0 docutils-0.17.1 furo-2022.2.23 imagesize-1.3.0 lazy-object-proxy-1.7.1 markdown-it-py-2.0.1 mdit-py-plugins-0.3.0 mdurl-0.1.0 myst-parser-0.17.0 nbsphinx-0.8.8 pbr-5.8.1 snowballstemmer-2.2.0 soupsieve-2.3.1 sphinx-autoapi-1.7.0 sphinx-autodoc-typehints-1.12.0 sphinx-automodapi-0.13 sphinx-click-3.1.0 sphinx-copybutton-0.3.3 sphinx-gallery-0.9.0 sphinx-inline-tabs-2021.3.28b7 sphinx-panels-0.6.0 sphinx-prompt-1.4.0 sphinx-togglebutton-0.2.3 sphinxcontrib-apidoc-0.3.0 sphinxcontrib-applehelp-1.0.2 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 unidecode-1.3.3 wrapt-1.13.3
> ```
> 
> I encountered a Pandoc missing error when building the docs with `python setup.py docs`:
> 
> ```
> Notebook error:
> PandocMissing in how_to/collection/query_electron_execution_result.ipynb:
> Pandoc wasn't found.
> Please check that pandoc is installed:
> https://pandoc.org/installing.html
> make: *** [html] Error 2
> ```
> 
> I'm guessing this is because I am setting up the doc for the first time on my Mac? And i need to install pandoc manually
> 
> Edit: I installed pandoc and the build succeeded (with some warnings). I can see the web pages in the build dir!

yes exactly. The Pandoc installation was what I was looking for.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -455,6 +501,42 @@ Some important commands to know about:
...
+Note that some additional packages might need to be installed as well, depending on the operating system.
+
+#### Building the docs
+
+Navigate to the root folder (`/covalent/`) of the repo and run:
+
+```bash
+python setup.py docs
+```
+
```
Comment (FyzHsn) [inline CONTRIBUTING.md:527]: Add the following section:

Note: You might need to install pandoc via `pip install pandoc`.

**Review by FyzHsn — CHANGES_REQUESTED** [review]

**Comment by Emmanuel289** [comment]
> > > > @HaimHorowitzAgnostiq I've added the latest changes. This is ready for your review.
> > > > @Emmanuel289 Adding you to this review as well. Let me know if there are any questions.
> > > 
> > > 
> > > @FyzHsn Thanks! I'll add in the list of new package installs on my Mac.
> > 
> > 
> > @HaimHorowitzAgnostiq @FyzHsn After running `pip install -e .`, the packages installed were
> > ```
> > Installing collected packages: e, cova
> >   Attempting uninstall: cova
> >     Found existing installation: cova 0.26.1
> >     Uninstalling cova-0.26.1:
> >       Successfully uninstalled cova-0.26.1
> > Successfully installed cova-0.27.2 e-1.4.5
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > After running `pip install -r tests/requirements`, the following packages were installed:
> > ```
> > Successfully built autograd future
> > Installing collected packages: webencodings, semantic-version, nodeenv, mistune, mccabe, ipython-genutils, iniconfig, distlib, appdirs, testpath, scipy, pyzmq, pyrsistent, pyflakes, pycodestyle, py, pluggy, pandocfilters, nest-asyncio, mock, jupyterlab-pygments, jupyter-core, isort, identify, future, filelock, entrypoints, defusedxml, coverage, cfgv, autoray, attrs, virtualenv, pytest, jupyter-client, jsonschema, flake8, bleach, autograd, pytest-mock, pytest-cov, pre-commit, pennylane, nbformat, nbclient, nbconvert
> > Successfully installed appdirs-1.4.4 attrs-21.4.0 autograd-1.3 autoray-0.2.5 bleach-4.1.0 cfgv-3.3.1 coverage-6.3.2 defusedxml-0.7.1 distlib-0.3.4 entrypoints-0.4 filelock-3.6.0 flake8-3.9.2 future-0.18.2 identify-2.4.11 iniconfig-1.1.1 ipython-genutils-0.2.0 isort-5.7.0 jsonschema-4.4.0 jupyter-client-7.1.2 jupyter-core-4.9.2 jupyterlab-pygments-0.1.2 mccabe-0.6.1 mistune-0.8.4 mock-4.0.3 nbclient-0.5.11 nbconvert-6.3.0 nbformat-5.1.3 nest-asyncio-1.5.4 nodeenv-1.6.0 pandocfilters-1.5.0 pennylane-0.16.0 pluggy-1.0.0 pre-commit-2.13.0 py-1.11.0 pycodestyle-2.7.0 pyflakes-2.3.1 pyrsistent-0.18.1 pytest-6.2.5 pytest-cov-2.12.0 pytest-mock-3.6.1 pyzmq-22.3.0 scipy-1.8.0 semantic-version-2.6.0 testpath-0.6.0 virtualenv-20.13.2 webencodings-0.5.1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > After running `pip install -r requirements.txt` in `/covalent/docs` (Building RTD locally), the following packages were installed:
> > ```
> > Successfully built autodocsumm sphinx-gallery lazy-object-proxy wrapt
> > Installing collected packages: snowballstemmer, alabaster, wrapt, unidecode, sphinxcontrib-serializinghtml, sphinxcontrib-qthelp, sphinxcontrib-jsmath, sphinxcontrib-htmlhelp, sphinxcontrib-devhelp, sphinxcontrib-applehelp, soupsieve, pbr, mdurl, lazy-object-proxy, imagesize, docutils, babel, Sphinx, markdown-it-py, beautifulsoup4, astroid, sphinxcontrib-apidoc, sphinx-togglebutton, sphinx-prompt, sphinx-panels, sphinx-inline-tabs, sphinx-gallery, sphinx-copybutton, sphinx-click, sphinx-automodapi, sphinx-autodoc-typehints, sphinx-autoapi, mdit-py-plugins, furo, autodocsumm, myst-parser, nbsphinx
> > Successfully installed Sphinx-4.0.3 alabaster-0.7.12 astroid-2.9.3 autodocsumm-0.2.6 babel-2.9.1 beautifulsoup4-4.10.0 docutils-0.17.1 furo-2022.2.23 imagesize-1.3.0 lazy-object-proxy-1.7.1 markdown-it-py-2.0.1 mdit-py-plugins-0.3.0 mdurl-0.1.0 myst-parser-0.17.0 nbsphinx-0.8.8 pbr-5.8.1 snowballstemmer-2.2.0 soupsieve-2.3.1 sphinx-autoapi-1.7.0 sphinx-autodoc-typehints-1.12.0 sphinx-automodapi-0.13 sphinx-click-3.1.0 sphinx-copybutton-0.3.3 sphinx-gallery-0.9.0 sphinx-inline-tabs-2021.3.28b7 sphinx-panels-0.6.0 sphinx-prompt-1.4.0 sphinx-togglebutton-0.2.3 sphinxcontrib-apidoc-0.3.0 sphinxcontrib-applehelp-1.0.2 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 unidecode-1.3.3 wrapt-1.13.3
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I encountered a Pandoc missing error when building the docs with `python setup.py docs`:
> > ```
> > Notebook error:
> > PandocMissing in how_to/collection/query_electron_execution_result.ipynb:
> > Pandoc wasn't found.
> > Please check that pandoc is installed:
> > https://pandoc.org/installing.html
> > make: *** [html] Error 2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I'm guessing this is because I am setting up the doc for the first time on my Mac? And i need to install pandoc manually
> > Edit: I installed pandoc and the build succeeded (with some warnings). I can see the web pages in the build dir!
> 
> yes exactly. The Pandoc installation was what I was looking for.

Awesome! Thanks

**Review by FyzHsn — APPROVED** [review]

**Pull request merged**